tests/conftest.py

```diff
@@ -479,8 +517,14 @@ def _exec_python_in_pod(kube_client, kube_ns, pod_name, code, kwargs=None, _retr
     print("Running {} in {}".format(code, pod_name))
     # need to create ws client to get returncode,
     # see https://github.com/kubernetes-client/python/issues/812
-    client = stream(
-        kube_client.connect_get_namespaced_pod_exec,
+    #
+    # That's why we are using the synchronous Kubernetes client here
+    # and why we imported them in the first place: kubernetes_asyncio
+    # does not yet support multichannel ws clients, which are needed
+    # to get the return code.
+    # cf https://github.com/tomplus/kubernetes_asyncio/issues/12
+    client = sync_stream(
```

Comment:
Great solution! I never thought of the fact that switching to kubernetes_asyncio didn't mean we couldn't still use the sync client for the missing feature. Makes perfect sense.

Reply:
Wish I could claim credit, but that idea on our end came from Russ Allbery (@rra)